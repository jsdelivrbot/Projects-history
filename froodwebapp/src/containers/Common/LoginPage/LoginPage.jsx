/* eslint-disable class-methods-use-this, jsx-a11y/label-has-for */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-flexbox-grid';
import { Button, Icon } from 'antd';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import { FormInput } from 'components';
import { reduxForm, Field } from 'redux-form';
import { loginRequest } from 'redux-base/actions/loginFlow';
// import loginFormValidation from './form/validation';
import { loginForm, loginButton } from './LoginPage.scss';

const mapStateToProps = state => ({
  user: state.login.user,
});

const mapDispatchToProps = {
  loginRequest,
  push
};

const reduxFormConfig = {
  form: 'loginPageForm',
  initialValues: {
    email: '',
    password: ''
  }
  // validate: loginFormValidation
};

export class LoginPage extends Component {
  componentWillReceiveProps(nextProps) {
    if (nextProps.user) {
      this.props.push('/loading');
    }
  }

  handleLogin = ({ email, password }) => {
    this.props.loginRequest(email, password);
  };

  render() {
    const {
      handleSubmit
    } = this.props;

    return (
      <form className={ loginForm }>
        <Row middle="xs">
          <Col xs md lg>
            <Field
              name="email"
              placeholder="Email"
              type="email"
              prefix={ <Icon type="user" style={ { fontSize: 13 } } /> }
              component={ FormInput }
            />
          </Col>
        </Row>
        <Row middle="xs">
          <Col xs md lg>
            <Field
              name="password"
              type="password"
              placeholder="Password"
              prefix={ <Icon type="lock" style={ { fontSize: 13 } } /> }
              component={ FormInput }
            />
          </Col>
        </Row>
        <Row middle="xs">
          <Col xs md lg>
            <Button
              className={ loginButton }
              type="primary"
              onClick={ handleSubmit(this.handleLogin) }
            >
              Log in
            </Button>
          </Col>
        </Row>
      </form>
    );
  }
}

LoginPage.propTypes = {
  // props
  user: PropTypes.object,
  // redux-base
  loginRequest: PropTypes.func.isRequired,
  // router
  push: PropTypes.func.isRequired,
  // redux-form related props
  handleSubmit: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm(reduxFormConfig)(LoginPage));
