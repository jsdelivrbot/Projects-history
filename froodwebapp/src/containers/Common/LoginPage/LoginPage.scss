.loginForm {
	display: block;
	transform: translate(-50%, -50%);
	top: 50%;
	left: 50%;
	position: absolute;
	width: 30vw;
	div[class*='row'] {
		margin: 2rem 0 0 0;
	}
}

.loginButton {
	width: 100%;
}