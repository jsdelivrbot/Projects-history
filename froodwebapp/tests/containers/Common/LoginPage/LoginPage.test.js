import { LoginPage } from 'containers/Common/LoginPage/LoginPage';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('LoginPage', () => {
  const props = {
    user: {},
    push: jest.fn(),
    loginRequest: jest.fn(),
    handleSubmit: jest.fn(),
  };

  it('renders the LoginPage component', () => {
    const component = shallow(<LoginPage { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of LoginPage', () => {
    const component = shallow(<LoginPage { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls loginRequest function in handleLogin function with correct param', () => {
    const user = {
      email: 'testuser28@gmail.com',
      password: '1234567'
    };
    const component = shallow(<LoginPage { ...props } />);
    component.instance().handleLogin(user);

    expect(props.loginRequest.mock.calls.length).toBe(1);

    expect(props.loginRequest.mock.calls[0][0]).toBe(user.email, user.password);
  });
});
