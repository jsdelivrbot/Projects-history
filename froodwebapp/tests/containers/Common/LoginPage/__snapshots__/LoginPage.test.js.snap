// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage capturing Snapshot of LoginPage 1`] = `
<form
  className="loginForm"
>
  <Row
    middle="xs"
  >
    <Col
      lg={true}
      md={true}
      xs={true}
    >
      <Field
        component={[Function]}
        name="email"
        placeholder="Email"
        prefix={
          <Icon
            style={
              Object {
                "fontSize": 13,
              }
            }
            type="user"
          />
        }
        type="email"
      />
    </Col>
  </Row>
  <Row
    middle="xs"
  >
    <Col
      lg={true}
      md={true}
      xs={true}
    >
      <Field
        component={[Function]}
        name="password"
        placeholder="Password"
        prefix={
          <Icon
            style={
              Object {
                "fontSize": 13,
              }
            }
            type="lock"
          />
        }
        type="password"
      />
    </Col>
  </Row>
  <Row
    middle="xs"
  >
    <Col
      lg={true}
      md={true}
      xs={true}
    >
      <Button
        className="loginButton"
        clicked={false}
        ghost={false}
        loading={false}
        prefixCls="ant-btn"
        type="primary"
      >
        Log in
      </Button>
    </Col>
  </Row>
</form>
`;
